x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpResult.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.http;
+
+import org.apache.http.HttpResponse;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.core.Streams;
+import org.elasticsearch.xpack.inference.common.SizeLimitInputStream;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Objects;
+
+public record HttpResult(HttpResponse response, byte[] body) {
+
+    public static HttpResult create(Settings settings, HttpResponse response) throws IOException {
+        return new HttpResult(response, limitBody(settings, response));
+    }
+
+    private static byte[] limitBody(Settings settings, HttpResponse response) throws IOException {
+        if (response.getEntity() == null) {
+            return new byte[0];
+        }
+
+        ByteSizeValue maxResponseSize = HttpSettings.MAX_HTTP_RESPONSE_SIZE.get(settings);
```

Comment:
If `maxResponseSize` is the only field read from the Settings pass the `ByteSizeValue` rather than the settings. 

For the settings to be updatable add a listener for the change. Search for examples of `clusterService.getClusterSettings().addSettingsUpdateConsumer(...)` in the code

Reply:
Actually this is harder than I thought it would be and my current solution will require the nodes to be restarted (or at least the inference plugin). This is because most of the settings (except the `maxResponseSize`) are all used when the http client is first initialized which only happens once when the plugin starts I believe.

Was your thought that this would only apply to the `maxResponseSize`? If it was for all the settings then we'll need a solution that listens for a change and actively closes and restarts the http client and connection evictor.

Reply:
I suppose a straightforward solution could be to create a wrapper class around the `HttpClient` and `HttpSettings` that listens for a settings change and then calls `HttpClient::close()` followed by `HttpClient::create()` and handles the synchronization. That'd probably cause any inflight requests to fail but that's probably ok and we can just document it.